### ISSUE TITLE
implication operator is left-associative

### ISSUE THREAD

**Issue by Jankoekenpan** [utwente-fmt/vercors#768]
Normally in formal logics the implication operator is right-associative.
e.g. `false ==> true ==> false` evaluates as `(false ==> (true ==> false))` which gives us `false ==> false`, then `true`.
VerCors 1.4.0 however treats the formula as `((false ==> true) ==> false)` which then reduces to `true ==> false`, then `false`.

Here is a repro file, it fails to verify on line 9 and 11 (6th and 8th assertion). The comments indicate the expected evaluation.
```
class Test {

    static void test() {
        assertTrue(true ==> true ==> true);     // [true ==> <true>],   [[<true>]]
        assertFalse(true ==> true ==> false);   // [true ==> <false>],  [[<false>]]
        assertTrue(true ==> false ==> true);    // [true ==> <true>],   [[<true>]]
        assertTrue(true ==> false ==> false);   // [true ==> <true>],   [[<true>]]
        assertTrue(false ==> true ==> true);    // [false ==> <true>],  [[<true>]]
        assertTrue(false ==> true ==> false);   // [false ==> <false>], [[<true>]]
        assertTrue(false ==> false ==> true);   // [false ==> <true>],  [[<true>]]
        assertTrue(false ==> false ==> false);  // [false ==> <true>],  [[<true>]]
    }

    requires b;
    static void assertTrue(boolean b);

    requires !b;
    static void assertFalse(boolean b);

}
```

**Comment by niomaster** [comment]
Thanks for the report! This is fixed in 2.0.0 onwards, and I'm a bit hesitant to change this in 1.4 now. Do you urgently require this for your case study?

**Comment by Jankoekenpan** [comment]
No, my current workaround is putting the parenthesis explicitly, and I don't have many double implications in my file, so I'm good for now.
